### PR TITLE
chore(ci): split build workflows by fedora version

### PR DESCRIPTION
**Pull request by bsherman**
This will allow us to see if a specific version of Fedora is building or not, plus allow rebuilding only a specific version.

Extra bonus: this will let us adjust merge checks to require success of a subset of versions, eg
"Check all 38 builds successful" and "Check all 39 builds successful" but not 40.

This is nice for pre-release builds like what we have now.

### CODE REVIEW

**Comment by bsherman** [comment]
This won't actually auto-merge, since I've changed the builds to run version specific "Check all NN builds successful" jobs instead of the single "Check all builds successful".

Once we get approvals, i'm happy to merge, or someone else can.

**Comment by dylanmtaylor** [comment]
Can this be used to remove a lot of the version-conditional logic in the reusable part by moving it into the version-specific files?

**Comment by EyeCantCU** [comment]
> Can this be used to remove a lot of the version-conditional logic in the reusable part by moving it into the version-specific files?

That's a great idea, but unfortunately, with how inheritance works in this instance, that's easier said than done and could lead to a bunch of redundant code

**Comment by bsherman** [comment]
> Can this be used to remove a lot of the version-conditional logic in the reusable part by moving it into the version-specific files?

As @EyeCantCU, I don't think it would help much in this case, and would probably cause other code duplication.

One thing I did contemplate moving, the conditionals for latest, stable and gts tagging... However, i chose to move them into the tag generation step (the only place they are used) and keep it there instead of passing in as inputs, because this keeps all the logic for it together, making it easier to read the whole context in one place.

**Review by EyeCantCU — APPROVED** [review]

File: .github/workflows/build-39.yml
Comment (noelmiller) [inline .github/workflows/build-39.yml:6]: Should this be set to 15?

File: .github/workflows/build-40.yml
Comment (noelmiller) [inline .github/workflows/build-40.yml:6]: This is set to 10?

Comment (bsherman) [reply .github/workflows/build-40.yml:6]: That's a fever error. I'll correct.

File: .github/workflows/reusable-build.yml
Comment (noelmiller) [inline .github/workflows/reusable-build.yml:105]: Should this be 38 again?

**Review by noelmiller — APPROVED** [review]

**Pull request merged**
